Automattic/jetpack
# Settings - Newsletter, Sharing: Disabling options for sites missing user connection

Fixes https://github.com/Automattic/jetpack-marketing/issues/1023

## Proposed changes:

* make sure that visually disabled elements receive HTML `disabled` property
* replace `unavailableInSiteConnectionMode` with `hasConnectedOwner` to enable options if the site has any connected user (also other than the current user)
* fix `save` button label when the button is disabled and not saving

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a JN instance from this branch
* Connect Jetpck but not the user
* Go to `/wp-admin/admin.php?page=jetpack#/newsletter` and you should see disabled controls that you can't toggle (not only CSS disabled)
* Do the same for the first option of `Sharing` settings section
* Create another user and connect that user
* Go back to the main user, and verify that you can change and save options in the Newsletter tab
* Verify that Sharing is also interactive

After the change, before connecting a user:
![SCR-20241004-obap](https://github.com/user-attachments/assets/3e72ff30-ae2c-4fb2-a900-95a03d02cf4b)
![SCR-20241004-njxe](https://github.com/user-attachments/assets/077f9ae5-6ea1-4195-a752-6649e8623ef2)
After connecting a user:
![SCR-20241004-nkxf](https://github.com/user-attachments/assets/79368e01-ce50-42e8-917d-6d9493645ef9)
